### PR TITLE
Add a product to pricebook

### PR DESCRIPTION
**Pull request by JuanCarlosMartinez4**

### CODE REVIEW

File: src/main/java/salesforce/ui/pages/opportunity/choosepricebook/AbstractChoosePriceBook.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+     * Selects price book by name.
+     * @param priceBookName value.
+     */
+    public abstract void selectPriceBookByName(String priceBookName);
+
+    /**
+     * Gets a web element.
+     * @param xpath value.
+     * @param concatText value.
+     * @return composed web element.
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/choosepricebook/AbstractChoosePriceBook.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/opportunity/AbstractOpportunityPage.java [hunk 2]
```diff
@@ -63,7 +64,7 @@
      * Sets the form of new Campaign.
      *
      * @param opportunity entity.
-     * @param fields map.
+     * @param fields      map.
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/AbstractOpportunityPage.java]: review this please

File: src/main/java/salesforce/ui/pages/opportunity/choosepricebook/AbstractChoosePriceBook.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @author Juan Martinez.
+ * @version 1.0 02 april 2020.
+ */
+public abstract class AbstractChoosePriceBook extends AbstractBasePage {
+
+    /**
+     * Selects price book by name.
+     * @param priceBookName value.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/choosepricebook/AbstractChoosePriceBook.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/opportunity/choosepricebook/ChoosePriceBookClassicPage.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+package salesforce.ui.pages.opportunity.choosepricebook;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+
+/**
+ * Manages opportunities price books page.
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/choosepricebook/ChoosePriceBookClassicPage.java]: choose

File: src/main/java/salesforce/ui/pages/opportunity/choosepricebook/ChoosePriceBookClassicPage.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Override
+    protected void waitUntilPageObjectIsLoaded() {
+
+    }
+
+    /**
+     * Gets price book selector web element.
+     * @return selector web element.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/choosepricebook/ChoosePriceBookClassicPage.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/opportunity/choosepricebook/ChoosePriceBookLightningPopup.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    private void clickOnPriceBookDropdownList() {
+        webDriverWait.until(ExpectedConditions.visibilityOf(priceBookDropdownList));
+        priceBookDropdownList.click();
+    }
+
+    /**
+     * Selects price book name.
+     * @param priceBookName value.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/choosepricebook/ChoosePriceBookLightningPopup.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/opportunity/opportunityproducts/AbstractOpportunityProductsPage.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * @author Juan Martinez.
+ * @version 1.0 02 april 2020.
+ */
+public abstract class AbstractOpportunityProductsPage extends AbstractBasePage {
+
+    /**
+     * Enables the option to select price book.
+     * @return choose price book instance.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/opportunityproducts/AbstractOpportunityProductsPage.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/product/newpricebookentry/AbstractNewPriceBookEntryPage.java [hunk 8]
```diff
@@ -18,4 +18,11 @@
  * @version 1.0 1 April 2020.
  */
 public abstract class AbstractNewPriceBookEntryPage extends AbstractBasePage {
+
+    /**
+     * Saves a new price book entity.
+     * @param listPriceValue string.
+     * @param priceBookName value.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/product/newpricebookentry/AbstractNewPriceBookEntryPage.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/product/newpricebookentry/NewPriceBookEntryLightningPopUp.java [hunk 9]
```diff
@@ -31,4 +32,27 @@
     protected void waitUntilPageObjectIsLoaded() {
 
     }
+
+    /**
+     * Sets list price book value.
+     * @param listPriceValue string.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/product/newpricebookentry/NewPriceBookEntryLightningPopUp.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/product/newproduct/AbstractNewProductPage.java [hunk 10]
```diff
@@ -60,6 +74,8 @@ public AbstractProductPage setProductInformation(final Product product, final Se
     protected HashMap<String, Runnable> composeStrategyMap(final Product product) {
         HashMap<String, Runnable> strategyMap = new HashMap<>();
         strategyMap.put(ProductConstant.NAME, () -> setProductName(product.getName()));
+        strategyMap.put(ProductConstant.CODE, () -> setProductCode(product.getCode()));
+        strategyMap.put(ProductConstant.DESCRIPTION, () -> setProductDescription(product.getDescription()));
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/product/newproduct/AbstractNewProductPage.java:78]: this I have already done

Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/ui/pages/product/newproduct/AbstractNewProductPage.java:78]: I think, but I needed to run the scenario

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 11]
```diff
@@ -367,4 +370,15 @@ public static AbstractProductPage getProductPage() {
         }
         return new ProductLightningPage();
     }
+
+    /**
+     * Gets new product instance.
+     * @return new product instance.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java]: ctrl + alt + L

File: src/test/java/salesforcetest/steps/ProductSteps.java [hunk 12]
```diff
@@ -13,7 +13,16 @@
 import io.cucumber.java.en.Then;
 import salesforce.entities.Context;
 import salesforce.entities.Product;
+import salesforce.ui.pages.AppPageFactory;
+import salesforce.ui.pages.genericTabs.AbstractTabObjectsPage;
+import salesforce.ui.pages.opportunity.AbstractOpportunityPage;
+import salesforce.ui.pages.product.AbstractProductPage;
+import salesforce.ui.pages.product.newpricebookentry.AbstractNewPriceBookEntryPage;
+import salesforce.ui.pages.product.newproduct.AbstractNewProductPage;
+
```
Comment (Amet015) [inline src/test/java/salesforcetest/steps/ProductSteps.java]: delete this line please

**Review by Amet015 — APPROVED** [review]

**Pull request merged**